capellambse/model/common/accessors.py

```diff
@@ -1769,9 +1769,7 @@ def __set__(
                     "Moving model objects is not supported yet"
                 )
             if self.__get__(obj) is not None:
-                raise NotImplementedError(
-                    "Replacing model objects is not supported yet"
-                )
+                obj._element.remove(obj._element.find(self.role_tag))
```

Comment:
We first need to deregister the XML subtree from the ID cache before actually removing anything.